sbi/inference/snl/snl.py

```diff
@@ -47,7 +47,9 @@ def __init__(
 
         if density_estimator is None:
             density_estimator = utils.likelihood_nn(
-                model="maf", prior=self._prior, x_o=self._x_o,
+                model="maf",
+                theta_dim=self._prior.sample().numel(),
```

Comment:
I am a bit confused by the fact that in the original code `x_o_dim` was actually a shape whereas `theta_dim` was a scalar, _provided theta had 2 dimensions, i.e. len(shape) ==2. I think if something is going to be `numel` then that should also be the name (i.e. why not `theta_numel`).